src/index.ts

```diff
@@ -0,0 +1,759 @@
+import axios from "axios";
+import { parse } from "node-html-parser";
+import {
+  District,
+  Game,
+  GameFilter,
+  GameState,
+  GetDistrict,
+  GetGame,
+  GetGames,
+  GetPlayer,
+  GetPlayers,
+  GetSchool,
+  GetSchools,
+  GetTeam,
+  GetTeams,
+  Player,
+  School,
+  SchoolType,
+  Sport,
+  Team,
+} from "./scraper";
+
+import puppeteer = require("puppeteer");
+
+export const getDistrict: GetDistrict = async (id): Promise<District> => {
+  const district = await axios.get(
+    "https://www.rankonesport.com/Schedules/View_Schedule_All_Web.aspx",
+    {
+      params: {
+        D: id,
+        MT: 0,
+      },
+    }
+  );
+
+  const dom = parse(district.data);
+
+  const districtName = dom.querySelector("#lbl_DistrictName")?.innerText ?? "";
+
+  const schoolIds: string[] = dom
+    .querySelectorAll("#schoolCont td a")
+    .map((schoolElement) => {
+      const href = schoolElement.getAttribute("href");
+      if (href) {
+        return new URLSearchParams(href).get("S");
+      }
+      return undefined;
+    })
+    .filter((schoolId) => {
+      return schoolId != null;
+    });
+
+  return {
+    id: id,
+    name: districtName,
+    schools: schoolIds,
+  };
+};
+
+export const getSchools: GetSchools = async (
+  districtId: string,
+  schoolType?: SchoolType
+): Promise<School[]> => {
+  const district = await axios.get(
+    "https://www.rankonesport.com/Schedules/View_Schedule_All_Web.aspx",
+    {
+      params: {
+        D: districtId,
+        MT: 0,
+      },
+    }
+  );
+
+  const dom = parse(district.data);
+
+  const highSchools = dom
+    .querySelectorAll("#schoolCont #highCont td a")
+    .map((schoolElement): [string, SchoolType] => {
+      const href = schoolElement.getAttribute("href");
+      if (href) {
+        return [new URLSearchParams(href).get("S"), "High"];
+      }
+      return undefined;
+    })
+    .filter((schoolId) => {
+      return schoolId != null;
+    });
+
+  const middleSchools: [string, SchoolType][] = dom
+    .querySelectorAll("#schoolCont #middleSchool td a")
+    .map((schoolElement): [string, SchoolType] => {
+      const href = schoolElement.getAttribute("href");
+      if (href) {
+        return [new URLSearchParams(href).get("S"), "Middle"];
+      }
+      return undefined;
+    })
+    .filter((schoolId) => {
+      return schoolId != null;
+    });
+
+  let schools: [string, SchoolType][] = [];
+
+  if (schoolType === "High") schools = highSchools;
+  else if (schoolType === "Middle") schools = middleSchools;
+  else {
+    schools = highSchools.concat(middleSchools);
+  }
+
+  const promises: Promise<School>[] = schools.map((school) => {
```

Comment:
what is this for?

Reply:
the promises const

Reply:
Lines 77-109 gather School IDs by parsing the HTML, the promises fetch data about those schools so the function can return an array of `School`s, not just school ids.

Reply:
ohhh that makes sense